CRM/Twingle/Form/Profile.php

```diff
@@ -327,11 +327,19 @@ public function buildQuickForm() {
       );
     }
 
+    $this->add(
+      'checkbox', // field type
+      'double_opt_in', // field name
+      E::ts('Use Double-Opt-In for newsletter'), // field label
+      FALSE, // is not required
+      array()
+      );
+
     $this->add(
       'select', // field type
       'newsletter_groups', // field name
       E::ts('Sign up for newsletter groups'), // field label
-      static::getNewsletterGroups(), // list of options
+      static::getNewsletterGroups($this->profile->getAttribute('double_opt_in')), // list of options
```

Comment:
The `double_opt_in` value must be passed to the `getNewsletterGroups()` function because in case that double opt-in is selected, the function must only give back **public** newsletter groups. Otherwise the `MailingEventSubscribe.create` will not work.